training/run_eval.py

```diff
@@ -556,22 +556,47 @@ def main():
 
         assistant_model.cuda()
 
-    # 6. Resample speech dataset: `datasets` takes care of automatically loading and resampling the audio,
+    # 6. Define generation arguments
+    gen_kwargs = {
+        "max_length": data_args.generation_max_length,
+        "return_timestamps": data_args.return_timestamps,
+        "num_beams": data_args.num_beams,
+        "top_k": 0,
+    }
+
+    if hasattr(model.generation_config, "is_multilingual") and model.generation_config.is_multilingual:
+        gen_kwargs["language"] = data_args.language
+        gen_kwargs["task"] = data_args.task
+    elif data_args.language is not None:
+        raise ValueError(
+            "Setting language token for an English-only checkpoint is not permitted. The language argument should "
+            "only be set for multilingual checkpoints."
+        )
```

Comment:
Informative error that will be thrown if `language` is passed but the checkpoint is English-only